### PR TITLE
SALTO-2974 include dependencies of translation collection instances i…

### PR DESCRIPTION
**Pull request by dantal4**
…n SDF deploy

Due to SDF bug, custcollections must be included in the project when referenced

---

_Additional context for reviewer_

---
_Release Notes_: 
Netsuite Adapter:
- include dependencies of translation collection instances in SDF deploy

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/reference_dependencies.ts [hunk 1]
```diff
@@ -148,12 +150,22 @@ const getRequiredReferencedElements = (
...
+  const referencedTranslationCollectionInstances = _.uniqBy(
+    await awu(elements)
+      .flatMap(findDependingElementsFromRefs)
+      .filter(isInstanceElement)
+      .filter(element => element.elemID.typeName === TRANSLATION_COLLECTION)
+      .filter(element => !elementsSet.has(elementFullName(element)))
+      .toArray(),
+    elementFullName
   )
+  return elements.concat(referencedTranslationCollectionInstances)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/reference_dependencies.ts:168]: Please move
``` log.debug(`adding referenced element:${os.EOL}${requiredReferencedElements.map(elementFullName).join('\n')}`)```
to here or add another log message.

**Review by omrilit — APPROVED** [review]
Looks good!
I am concerned that this code runs after the filters, and not covered in the changeValidators.
I think that we must move this logic to run in a proper place.
Now with the server-validations we will force the user to add these instances to their deploy plan.
In some cases, there will be no change in the referenced instance which may make it not-selectable and require turning off the server validations.

In addition, the fact that we deploy elements that the user is not aware of (we don't require these references to be deployed in an implicit way), is pretty severe.
We should either do it as part of https://salto-io.atlassian.net/browse/SAAS-6379, or somehow notify to the user in a changeValidator (?) that it's about to be done.

I am approving since your code is just another example for an existing behavior.
But I think that we should tackle it now. WDYT? I will open a ticket.

**Comment by omrilit** [comment]
I opened https://salto-io.atlassian.net/browse/SALTO-2975

**Comment by dantal4** [comment]
@omrilit A small correction - it is partially covered by change validators in these validators:
- `clientValidation` - the dependencies are added in `client.ts` in both deploy & validate flows
- `safeDeploy` - all additional instances are taken care too.

**Comment by omrilit** [comment]
OK that's a good point, so it means that it's not related to the server-validations, but is still done implicitly and may cause some confusion and lack of confidence in Salto for our users.
I will update in https://salto-io.atlassian.net/browse/SALTO-2975

**Pull request merged**